src/Unread.ts

```diff
@@ -159,15 +159,13 @@ function makeHasReceipt(
         // If we found an event matching our receipt, then it's easy: this event
         // has a receipt if its ID is the same as the one in the receipt.
         return (ev) => ev.getId() == readUpToId;
-    } else {
-        // If we didn't, we have to guess by saying if this event is before the
-        // receipt's ts, then it we pretend it has a receipt.
-        const receipt = roomOrThread.getReadReceiptForUserId(myUserId);
-        if (receipt) {
-            const receiptTimestamp = receipt.data.ts;
-            return (ev) => ev.getTs() < receiptTimestamp;
-        } else {
-            return (_ev) => false;
-        }
     }
+
+    // If we didn't, we have to guess by saying if this event is before the
+    // receipt's ts, then it we pretend it has a receipt.
+    const receiptTs = roomOrThread.getReadReceiptForUserId(myUserId)?.data.ts ?? 0;
```

Comment:
Ohhh good catch here, I was running under the assumption that  `getReadReceiptForUserId` was returning the unthreaded receipt too. 

That'll definitely get rid of quite a few of the unreads!